### PR TITLE
Add new rook subscription

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
Added rook-ceph-operator subscription to odf-operator Also, updated the odf-catalogsource to include rook-ceph-operator bundle in it.

### CODE REVIEW

**Review by iamniting — CHANGES_REQUESTED** [review]
We also need to add it in the OLM deps inside bundle.

File: controllers/subscriptions.go [hunk 1]
```diff
@@ -385,7 +385,32 @@ func GetStorageClusterSubscriptions() []*operatorv1alpha1.Subscription {
...
+						Operator: "Equal",
+						Value:    "true",
+						Effect:   "NoSchedule",
+					},
+				},
+			},
+		},
+	}
+
+	return []*operatorv1alpha1.Subscription{noobaaSubscription, ocsSubscription, csiAddonsSubscription, rookSubscription}
```
Comment (iamniting) [inline controllers/subscriptions.go:413]: Lets reorder this, 
`ocs, rook, noobaa, csi`

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/subscriptions.go [hunk 2]
```diff
@@ -223,7 +223,7 @@ func GetVendorCsvNames(kind odfv1alpha1.StorageKind) []string {
 	if kind == VendorFlashSystemCluster() {
 		csvNames = []string{IbmSubscriptionStartingCSV}
 	} else if kind == VendorStorageCluster() {
-		csvNames = []string{NoobaaSubscriptionStartingCSV, OcsSubscriptionStartingCSV, CSIAddonsSubscriptionStartingCSV}
+		csvNames = []string{NoobaaSubscriptionStartingCSV, OcsSubscriptionStartingCSV, CSIAddonsSubscriptionStartingCSV, RookSubscriptionStartingCSV}
```
Comment (iamniting) [inline controllers/subscriptions.go:226]: reorder this also
`ocs, rook, noobaa, csi`

**Review by iamniting — CHANGES_REQUESTED** [review]

**Comment by Nikhil-Ladha** [comment]
Tested the changes and everything is working fine

File: bundle/metadata/dependencies.yaml [hunk 3]
```diff
@@ -11,3 +11,7 @@ dependencies:
   value:
     packageName: csi-addons
     version: "0.7.0"
+- type: olm.package
```
Comment (iamniting) [inline bundle/metadata/dependencies.yaml:14]: Lets move it after ocs

File: controllers/subscriptions.go [hunk 4]
```diff
@@ -385,7 +385,32 @@ func GetStorageClusterSubscriptions() []*operatorv1alpha1.Subscription {
...
+			Channel:                RookSubscriptionChannel,
+			StartingCSV:            RookSubscriptionStartingCSV,
+			InstallPlanApproval:    operatorv1alpha1.ApprovalAutomatic,
+			Config: &operatorv1alpha1.SubscriptionConfig{
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      "node.ocs.openshift.io/storage",
+						Operator: "Equal",
+						Value:    "true",
+						Effect:   "NoSchedule",
```
Comment (iamniting) [inline controllers/subscriptions.go:406]: It is better to have it as part of the CSV itself.

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/subscriptions.go (hunk 4 above)
Comment (Nikhil-Ladha) [reply controllers/subscriptions.go:406]: cc @subhamkrai please take care of this in the rook PR

**Review by iamniting — CHANGES_REQUESTED** [review]
pls make the e2e tests changes as well as a separate commit.

**Review by iamniting — APPROVED** [review]

**Pull request merged**